### ISSUE TITLE
Cop to detect a space between Hash and [key]

### ISSUE THREAD

**Issue by yahonda** [rubocop-hq/rubocop#9305]
## Is your feature request related to a problem? Please describe.

I'd like RuboCop to detect a space between the hash object and [] method.

```ruby
$ more foo.rb
# frozen_string_literal: true

foo = { bar: 10 }

# bad
p foo [:bar]
# Detected by `Layout/ExtraSpacing` cop
p foo  [:bar]
# good
p foo[:bar]
```

```ruby
$ ruby -v
ruby 3.0.0p0 (2020-12-25 revision 95aff21468) [x86_64-linux]
$ rubocop -V
warning: parser/current is loading parser/ruby30, which recognizes
warning: 3.0.0-dev-compliant syntax, but you are running 3.0.0.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
1.7.0 (using Parser 2.7.2.0, rubocop-ast 1.3.0, running on ruby 3.0.0 x86_64-linux)
$ rubocop foo.rb
warning: parser/current is loading parser/ruby30, which recognizes
warning: 3.0.0-dev-compliant syntax, but you are running 3.0.0.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
Inspecting 1 file
C

Offenses:

foo.rb:8:6: C: [Correctable] Layout/ExtraSpacing: Unnecessary spacing detected.
p foo  [:bar]
     ^

1 file inspected, 1 offense detected, 1 offense auto-correctable
$
```

## Describe the solution you'd like
I'd like RuboCop to detect a space between the hash object and [] method.

## Describe alternatives you've considered
If there are two spaces between Hash and [key], `Layout/ExtraSpacing` cop detects it.

**Comment by koic** [comment]
You can detect it with `Layout/SpaceBeforeBrackets` cop. 
https://docs.rubocop.org/rubocop/1.7/cops_layout.html#layoutspacebeforebrackets

The cop has been added in RuboCop 1.7.0, but please try using a newer next version (1.7.1 or 1.8.0) that will be released due to existing issue.
https://github.com/rubocop-hq/rubocop/pull/9291

Thank you.

**Comment by yahonda** [comment]
Thanks for the information. I have installed the latest RuboCop and confirmed a space between hash and [] has been detected by `Layout/SpaceBeforeBrackets` cop.


```
$ more foo.rb
# frozen_string_literal: true

foo = { bar: 10 }

# bad
p foo [:bar]
# Detected by `Layout/ExtraSpacing` cop
p foo  [:bar]
# good
p foo[:bar]
$ more Gemfile
gem 'rubocop', git: 'https://github.com/rubocop-hq/rubocop'
$ bundle install
$ bundle exec rubocop -V
1.7.0 (using Parser 3.0.0.0, rubocop-ast 1.3.0, running on ruby 2.7.2 x86_64-linux)
$ bundle exec rubocop --only Layout/SpaceBeforeBrackets foo.rb -a
Inspecting 1 file
C

Offenses:

foo.rb:6:6: C: [Corrected] Layout/SpaceBeforeBrackets: Remove the space before the opening brackets.
p foo [:bar]
     ^
foo.rb:8:6: C: [Corrected] Layout/SpaceBeforeBrackets: Remove the space before the opening brackets.
p foo  [:bar]
     ^^

1 file inspected, 2 offenses detected, 2 offenses corrected
$
```
